beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -360,64 +402,108 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
             verify_kzg_for_blob_list(all_blobs.iter(), kzg)?;
         }
 
+        let all_data_columns = blocks
+            .iter()
+            .filter(|block| self.data_columns_required_for_block(block.as_block()))
+            // this clone is cheap as it's cloning an Arc
+            .filter_map(|block| block.custody_columns().cloned())
+            .flatten()
+            .map(CustodyDataColumn::into_inner)
+            .collect::<Vec<_>>();
```

Comment:
we can probably add a condition here to avoid the allocation for the vec and list below if peerdas is not enabled

Reply:
ah ignore this for now - this is a `Vec` of rpc blocks, so we've gotta check every block.